### ISSUE TITLE
data type not inexact (integers support)

### ISSUE THREAD

**Issue by ayoubft** [observingClouds/xbitinfo#208]
When setting the `mask_and_scale` argument to `False`, the data variables are **integers**:

<img src='https://github.com/observingClouds/xbitinfo/assets/63267601/5bf0a071-1d27-440e-b06b-2a8180d307a4' width=400px>

When trying to get the bitinformation with:
```python
info_per_bit = xb.get_bitinformation(ds, dim="latitude", implementation="python")
```
The error below is raised, as integer formats are not yet supported, there have been some efforts made in this direction (#206), but it is unclear whether the integer formats should have their own logic or if we can modify the existing code for floats.

The error:
```python
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[6], line 2
      1 # get information content per bit
----> 2 info_per_bit = xb.get_bitinformation(ds, dim="latitude", implementation="python")

File ~/Projects/xbitinfo/xbitinfo/xbitinfo.py:236, in get_bitinformation(ds, dim, axis, label, overwrite, implementation, **kwargs)
    234         info_per_bit[var] = info_per_bit_var
    235 elif implementation == "python":
--> 236     info_per_bit_var = _py_get_bitinformation(ds, var, axis, dim, kwargs)
    237     if info_per_bit_var is None:
    238         continue

File ~/Projects/xbitinfo/xbitinfo/xbitinfo.py:296, in _py_get_bitinformation(ds, var, axis, dim, kwargs)
    294 astype = f"u{itemsize}"
    295 X = da.array(ds[var])
--> 296 X = pb.signed_exponent(X)
    297 X = X.astype(astype)
    298 if axis is not None:

File ~/Projects/xbitinfo/xbitinfo/_py_bitinfo.py:72, in signed_exponent(A)
     69 uinttype = f"u{itemsize}"
     70 inttype = f"i{itemsize}"
---> 72 sign_mask = 1 << np.finfo(A.dtype).bits - 1
     73 sfmask = sign_mask | (1 << np.finfo(A.dtype).nmant) - 1
     74 emask = exponent_mask(A.dtype)

File ~/miniconda3/envs/bitinfo/lib/python3.11/site-packages/numpy/core/getlimits.py:492, in finfo.__new__(cls, dtype)
    490     dtype = newdtype
    491 if not issubclass(dtype, numeric.inexact):
--> 492     raise ValueError("data type %r not inexact" % (dtype))
    493 obj = cls._finfo_cache.get(dtype, None)
    494 if obj is not None:

ValueError: data type <class 'numpy.int16'> not inexact
```

**Comment by milankl** [comment]
duplicate of #206, adressed in #215

**Comment by observingClouds** [comment]
Fixed by #215
